shogun-lib/src/main/java/de/terrestris/shogun/lib/security/access/entity/BaseEntityPermissionEvaluator.java

```diff
@@ -230,4 +242,80 @@ public boolean hasPermissionByGroupClassPermission(User user, BaseEntity entity,
         return groupClassPermissions.contains(permission) ||
             groupClassPermissions.contains(PermissionType.ADMIN);
     }
+
+    /**
+     * Default <code>findAll</code> implementation which supports pagination.
+     * Speeds up the permission check by utilizing two simplifications:
+     * 1) If the authenticated user has role `ADMIN` or has class-level permission it skips further permission checks.
+     * 2) Otherwise, user and group instance permissions are checked while querying the data. This removes the need for
+     * any additional filtering.
+     *
+     * @param user The authenticated user.
+     * @param pageable The pagination configuration.
+     * @param repository The base entity repository used to fetch the entities.
+     * @return A page of entities.
+     */
+    @Override
+    public Page<E> findAll(User user, Pageable pageable, BaseCrudRepository<E, Long> repository, Class<E> baseEntityClass) {
+        // option A: user has role `ADMIN`
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        List<GrantedAuthority> authorities = new ArrayList<>(authentication.getAuthorities());
+        var isAdmin = authorities.stream().anyMatch(
+            grantedAuthority -> StringUtils.endsWithIgnoreCase(grantedAuthority.getAuthority(), "ADMIN")
```

Comment:
An `endsWith` check might result in false positives, I would suggest to check against the full role name with `equalsIgnoreCase`.

Reply:
fixed in 63741cd4c7a03a279856fdbdf9e73df0cd855a95